Game : Stop The Breed

lunayach opened the pull request:
### Description
Implemented the prevention game, Stop The Breed.

Fixes #120 

### Type of Change:


- Code
- Quality Assurance
- User Interface
- Outreach


**Code/Quality Assurance Only**
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)



### How Has This Been Tested?
Both Karma and Mocha tests passed.


### Checklist:


- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

shreyans29 commented on src/app/app.module.ts:
```diff
@@ -40,6 +38,9 @@ import { StageCompleteComponent } from './stage-complete/stage-complete.componen
 import { DoctorchatComponent } from './meds-n-labels/activity-3/activity-3.component';
 import { UnlockedStageComponent } from './unlocked-stage/unlocked-stage.component';
 import { HowToPlayComponent } from './unlocked-stage/how-to-play/howtoplay.component';
+import {PreventionComponent} from './prevention/prevention.component';
```
spacing again..

shreyans29 commented on src/app/app.routing.ts:
```diff
@@ -148,7 +150,22 @@ export const routes: Routes = [
         canActivate: [LoggedInGuard]
-    }
+    },
+  {
```
spacing is off here..

shreyans29 replied:
Hey @lunayach this needs to be corrected..

shreyans29 commented on src/app/material.module.ts:
```diff
@@ -1,19 +1,27 @@
 import { NgModule } from '@angular/core';
-import { MatToolbarModule } from '@angular/material/toolbar';
-import { MatButtonModule } from '@angular/material';
+import {MatToolbarModule} from '@angular/material/toolbar';
```
Check for similar issue in the whole PR , it is in a lot of places.

shreyans29 commented on src/app/prevention/prevention.component.scss:
```diff
@@ -0,0 +1,5 @@
+@import "../styles/main";
+
+.content {
+  margin-bottom: 100px;
```
check if 100 px is already defined as a variable.. if not define it in the scss structure and then use.

shreyans29 commented on src/app/prevention/stop-the-breed/game-config.ts:
```diff
@@ -0,0 +1,79 @@
+export const backgroundConfig: Item = {
+  imageURL: '../../../assets/img/stop-the-breed/background_village.png',
+  positionX: 0,
+  positionY: 0,
```
Maintain a configuration file where these co-ordinates are defined. It will be easy to change when in the file when you want to change. Do not hard code.

shreyans29 replied:
not just coordinates, the url and everything which can be changed.

lunayach replied:
@shreyans29 this is itself the game-config file. I made this for that purpose only. Do I need to make a config file for this as well?

paavininanda commented on src/app/prevention/stop-the-breed/stop-the-breed.module.ts:
```diff
@@ -0,0 +1,13 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import {StopTheBreedComponent} from './stop-the-breed.component';
```
Spacing issues here too.

paavininanda commented on src/app/shared.module.ts:
```diff
@@ -0,0 +1,12 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import {ButtonNavComponent} from './button-nav/button-nav.component';
```
Spacing.

Raxerz commented on package.json:
```diff
@@ -43,7 +44,6 @@
     "handlebars": "^4.0.10",
     "hbs": "~4.0.1",
     "intl": "^1.2.5",
-    "jasmine": "^3.1.0",
```
not required?

Raxerz commented on package.json:
```diff
@@ -84,15 +85,15 @@
     "jasmine-spec-reporter": "~4.1.0",
-    "karma": "^1.7.1",
-    "karma-chrome-launcher": "^2.1.1",
+    "karma": "~1.7.0",
```
why is karma downgraded

Raxerz commented on package.json:
```diff
@@ -84,15 +85,15 @@
     "karma-cli": "~1.0.1",
     "karma-coverage-istanbul-reporter": "^1.2.1",
-    "karma-jasmine": "^1.1.2",
+    "karma-jasmine": "~1.1.0",
```
why downgrade?

Raxerz commented on src/app/prevention/stop-the-breed/game-config.ts:
```diff
@@ -0,0 +1,79 @@
+    height: 90,
+    messageText: 'They are potential breeding grounds for Mosquitoes. ' +
+      'Do look for them in your surroundings and remove them.',
+    messageImageURL: '../../../assets/img/stop-the-breed/message/tyre.jpg'
```
The relative paths for the URL may work fine but I think this would cause problems in future if the folder structure changes. Maybe we can split the url in such a way that making any changes to the folder structure will require no or minimal changes to the files. 
In this case the path should be changed at each and every place it is being referenced.

lunayach replied:
Yes! Angular already provides the way for that. We had previously defined our 'assets' folder in the 'assets' property of Angular.json Didn't realise it earlier. Have pushed the required changes.

Raxerz commented:
@lunayach can the image be aligned to the center? 
![prevention](https://user-images.githubusercontent.com/16686007/41050434-fcd30074-69d0-11e8-8161-be599b4efa43.png)

Also you have not added the prevention stage in the menu page and the buttons are still square, wasn't it fixed in the previous PR?
![button](https://user-images.githubusercontent.com/16686007/41050666-ae35f2b8-69d1-11e8-843a-2c4402365430.png)

lunayach commented:
@Raxerz 
Okay, would centre the image.
Actually had completely forgotten about the Menu thing. Would do that.
And yes! this square button issue was fixed in previous PR. See this from my dev branch. 
![image](https://user-images.githubusercontent.com/21227893/41051290-51da6a10-69d3-11e8-96a7-e347df02aa01.png)
Weired that, this behaviour is changing after adding that game. Would resolve that.

lunayach commented:
@Raxerz Have addressed all the issues. Though for now the image for prevention that comes in main-menu, I have used the already-used one. Have mentioned this issue on slack.

lunayach commented:
@Raxerz have put an image as per slack discussions. Please check. :+1:
